src/components/SchemaRow/SchemaRow.tsx

```diff
@@ -80,15 +84,49 @@ export const SchemaRow: React.FunctionComponent<SchemaRowProps> = ({ schemaNode,
                 />
               ) : null}
 
-              {schemaNode.subpath.length > 0 &&
-                isCombiner(schemaNode.subpath[0]) &&
-                schemaNode.parent?.children?.indexOf(schemaNode as any) !== 0 && (
-                  <Divider kind={schemaNode.subpath[0]} />
+              <div className="sl-flex sl-items-center sl-text-base sl-flex-1 sl-truncate">
+                {schemaNode.subpath.length > 0 && shouldShowPropertyName(schemaNode) && (
+                  <div className="sl-mr-2 sl-font-mono sl-font-bold">{last(schemaNode.subpath)}</div>
+                )}
+
+                {choices.length === 1 && (
+                  <>
+                    <Types schemaNode={typeToShow} />
+                    <Format schemaNode={typeToShow} />
+                  </>
                 )}
 
-              <div className="sl-flex sl-text-base sl-flex-1 sl-truncate">
-                <Property schemaNode={schemaNode} />
-                <Format schemaNode={schemaNode} />
+                {onGoToRef && isReferenceNode(schemaNode) && schemaNode.external ? (
+                  <a
+                    className="sl-ml-2 sl-cursor-pointer sl-text-primary-light"
+                    onClick={(e: React.MouseEvent) => {
+                      e.preventDefault();
+                      e.stopPropagation();
+                      onGoToRef(schemaNode);
+                    }}
+                  >
+                    (go to ref)
+                  </a>
+                ) : null}
+
+                {schemaNode.subpath.length > 1 && schemaNode.subpath[0] === 'patternProperties' ? (
+                  <div className="sl-ml-2 sl-truncate sl-text-muted">(pattern property)</div>
+                ) : null}
+                {choices.length > 1 && (
+                  <Select
+                    size="sm"
+                    options={choices.map((choice, index) => ({
+                      value: index,
```

Comment:
So perhaps let's just do `String(index)`
```suggestion
                      value: String(index), // or toString()
```